crates/xlayer/db/src/internal_transaction_inspector.rs

```diff
@@ -0,0 +1,307 @@
+    sibling_count: Vec<usize>,
+    trace_stack: Vec<usize>,
+}
+
+impl Default for TraceCollector {
+    fn default() -> Self {
+        Self {
+            all_traces: Vec::<Vec<InternalTransaction>>::default(),
+            traces: Vec::<InternalTransaction>::default(),
+            current_path: Vec::<usize>::default(),
+            last_depth: 0,
+            sibling_count: vec![0],
+            trace_stack: Vec::<usize>::default(),
+        }
+    }
+}
+
+impl TraceCollector {
+    fn format_error(result: &reth_revm::interpreter::InstructionResult) -> String {
+        use reth_revm::interpreter::InstructionResult;
+
+        match result {
+            InstructionResult::Revert => "execution reverted".to_string(),
+            // InstructionResult::OutOfGas => "out of gas".to_string(),
+            // InstructionResult::OutOfFund => "insufficient funds".to_string(),
+            // InstructionResult::CallTooDeep => "call depth exceeded".to_string(),
+            // InstructionResult::OpcodeNotFound => "invalid opcode".to_string(),
+            // InstructionResult::InvalidJump => "invalid jump destination".to_string(),
+            // InstructionResult::InvalidFEOpcode => "invalid 0xfe opcode".to_string(),
+            // InstructionResult::StackOverflow => "stack overflow".to_string(),
+            // InstructionResult::StackUnderflow => "stack underflow".to_string(),
+            // InstructionResult::OutOfOffset => "out of offset".to_string(),
+            // InstructionResult::CreateCollision => "contract address collision".to_string(),
+            // InstructionResult::OverflowPayment => "payment overflow".to_string(),
+            // InstructionResult::PrecompileError => "precompile error".to_string(),
+            // InstructionResult::NonceOverflow => "nonce overflow".to_string(),
+            // InstructionResult::CreateContractSizeLimit => "contract size limit exceeded".to_string(),
+            // InstructionResult::CreateContractStartingWithEF => "contract starts with 0xEF".to_string(),
+            // InstructionResult::CreateInitCodeSizeLimit => "init code size limit exceeded".to_string(),
+            _ => format!("{:?}", result),
+        }
+    }
+
+    fn init_op(
+        &mut self,
+        call_type: String,
+        from: String,
+        to: String,
+        input: Bytes,
+        value_wei: String,
+        gas_limit: u64,
+        code_address: String,
+    ) {
+        self.traces.push(InternalTransaction::default());
+
+        let txn = self.traces.last_mut().unwrap();
+        txn.call_type = call_type;
+        txn.from = from.clone();
+        txn.input = input;
+        txn.is_error = false;
+        txn.gas = gas_limit;
+        txn.value_wei = if value_wei.is_empty() { "0" } else { &value_wei }.to_string();
+        txn.call_value_wei = if let Ok(value) = value_wei.parse::<u128>() {
+            format!("0x{:x}", value)
+        } else {
+            String::from("0x0")
+        };
```

Comment:
```suggestion
        txn.call_value_wei = match value_wei.parse::<u128>() {
            Ok(value) => format!("0x{:x}", value),
            _ => String::from("0x0"),
        };
```
`match` looks cleaner.